Update statistics.md and convert to RST

jasonblais opened the pull request.

esethna commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,75 @@
+your system, including groups that may have been archived.
+
+**Active Users With Posts:** Users who logged in and made a post on a
+certain day in all the teams on your system.
```
`... certain day in all the teams` > `... certain day in any of the teams on your system.`

esethna commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,75 @@
+browser activity in Mattermost.
+
+**Newly Created Users:** Users that have recently completed the sign-up
+process to create a Mattermost account on the team.
```
I think this might need to be clarified. ie is it new joins on the team? Since users can now have accounts that aren't connected to teams

esethna commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,75 @@
+-----------------
+
+System statistics are viewable under **System Console** > **Site
+Reports** > **View Statistics**. The data shown here is a cumulative sum
```
System Console > Reporting > Site Statistics

esethna commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,75 @@
+Statistics on users, posts and channels are tracked for each system and
+team. Enterprise Editions have access to advanced system statistics.
+
+System Statistics
```
Site Statistics

esethna commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,75 @@
+
+**Total Users:** The total number of accounts created on your system,
+including both active and inactive accounts.
+
```
Missing Total Teams

esethna commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,75 @@
+**Public Groups:** The number of public channels created in all the
+teams on your system, including channels that may have been archived.
+
+**Private Groups:** The number of private groups in all the teams on
```
Public Channels and Private Groups are now just "Total Channels"

esethna commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,75 @@
+
+**Private Groups:** The number of private groups in all the teams on
+your system, including groups that may have been archived.
+
```
Missing Total Sessions, Total Commands, Websocket Conns, Master DB Conns, Replica DB Conns

esethna commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,75 @@
+
+**Active Users With Posts:** Users who logged in and made a post on a
+certain day in all the teams on your system.
+
```
Missing Total Posts graph

esethna commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,75 @@
+**Total Posts:** The total number of posts made in your team, including
+deleted posts and posts made using automation.
+
+**Public Groups:** The number of public channels created by your team,
```
Public Channels

esethna commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,75 @@
+**Active Users With Posts:** Users who logged in and made a post on a
+certain day. Includes system messages posted from the user's account.
+
+**Recently Active Users:** Users that have logged in and had recent
```
Recently > Recent

jasonblais commented:
Thanks! Updated

jasonblais commented:
It would probably be worth for a dev (maybe @grundleborg) to review that the technical details of the field descriptions are correct. E.g. `**Total Sessions:** The number of active user sessions connected to your system. Expired sessions are not counted.`

it33 commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Active Users With Posts (graph):** Users who logged in and made a post on a certain day in all the teams on your system, including system messages posted from the user's account.
+
+Advanced system statistics (Enterprise)
```
Propose: Advanced system statistics (Enterprise) > Advanced System Statistics (Enterprise)

jasonblais replied:
Section titles should be sentence case per our style guide  https://docs.mattermost.com/process/sg_mattermost-doc-style.html#capitalization

I changed "Site Statistics" and "Team Statistics" should be sentence case too.

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+System statistics are viewable under **System Console** > **Reporting** > **Site Statistics**. The data shown here is a cumulative sum
+across all teams on the system.
+
+**Total Users:** The total number of accounts created on your system, including both active and inactive accounts.
```
If "inactive" means DeleteAt > 0 then this description is incorrect. The System Console shows the number of users on the system who's accounts have not been deleted.

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Total Users:** The total number of accounts created on your system, including both active and inactive accounts.
+
+**Total Teams:** The total number of teams created on your system, including deleted teams.
```
This does not include deleted teams.

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Total Posts:** The total number of posts made in all the teams on your system, including deleted posts and posts made using automation.
+
+**Total Channels:** The total number of channels created in all the teams on your system, including deleted channels.
```
This is Public + Private channels (not including Direct ones). Perhaps this should be explained explicitly?

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Total Posts (graph):** The total number of posts made on a certain day in all the teams on your system, including deleted posts and posts made using automation.
+
+**Active Users With Posts (graph):** Users who logged in and made a post on a certain day in all the teams on your system, including system messages posted from the user's account.
```
I wonder if it's a bit confusing saying "...logged in and made a post on a certain day...", since really it's just made a post. They didn't specifically log in on that day, and by saying that it seems like this is an important part of the criteria to appear on that graph.

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+Team Statistics are viewable under **System Console** > **Teams** > **Statistics**.
+
+**Total Users:** The total number of accounts created on your system, including both active and inactive accounts.
```
Doesn't actually include inactive accounts. Also this should say something about "who are members of this team".

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Total Users:** The total number of accounts created on your system, including both active and inactive accounts.
+
+**Public Channels:** The number of public channels created by your team, including channels that may have been archived.
```
Does not include archived ones.

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Public Channels:** The number of public channels created by your team, including channels that may have been archived.
+
+**Private Groups:** The number of private groups created by your team, including groups that may have been archived.
```
Does not include archived ones.

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Private Groups:** The number of private groups created by your team, including groups that may have been archived.
+
+**Total Posts:** The total number of posts made in all the teams on your system, including deleted posts and posts made using automation.
```
This is the count for a given team, so does not include all teams on your system.

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Total Posts:** The total number of posts made in all the teams on your system, including deleted posts and posts made using automation.
+
+**Total Posts (graph):** The total number of posts made on a certain day in all the teams on your system, including deleted posts and posts made using automation.
```
Just the posts for that team.

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Total Posts (graph):** The total number of posts made on a certain day in all the teams on your system, including deleted posts and posts made using automation.
+
+**Active Users With Posts (graph):** Users who logged in and made a post on a certain day in all the teams on your system, including system messages posted from the user's account.
```
Just for that team. Also, same comment for this one as in "System Analytics" section: I wonder if it's a bit confusing saying "...logged in and made a post on a certain day...", since really it's just made a post. They didn't specifically log in on that day, and by saying that it seems like this is an important part of the criteria to appear on that graph.

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Active Users With Posts (graph):** Users who logged in and made a post on a certain day in all the teams on your system, including system messages posted from the user's account.
+
+**Recent Active Users:** Users that have logged in and had recent browser activity in Mattermost.
```
Just for this team. Also you may want to specify that this is the 100 most recently active users?

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Recent Active Users:** Users that have logged in and had recent browser activity in Mattermost.
+
+**Newly Created Users:** Users that have recently joined the team.
```
I'm actually a bit suspicious of this one, as the code appears to just return the most recently created n users from the list of recently active users, which doesn't really fit any definition of "Newly Created Users" very accurately.

grundleborg commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,66 @@
+
+**Channel Types:** This chart displays the number of public channels and private groups in a visual format, including channels that may have been archived.
+
+**Posts, Files and Hashtags:** This chart displays the number of posts containing files, hashtags or text-only. Posts containing both files and hashtags are counted in both categories.
```
Deleted posts are included.

jasonblais commented on source/administration/statistics.rst:
```diff
@@ -0,0 +1,75 @@
+**Posts, Files and Hashtags:** This chart displays the number of posts
+containing files, hashtags or text-only. Posts containing both files and
+hashtags are counted in both categories.
+
```
@it33 They're under "Advanced EE Stats"

jasonblais commented:
@grundleborg Replying separately, since your comments are now shown as "outdated".

First, excellent work verifying this data, this is extremely useful! Looks like some of our definitions weren't really accurate at all.

For "Recent active users", does it list the 100 most recent active users up to a certain time period? For instance [on ci-windows-mysql, for team "Mattermost"](https://ci-windows-mysql.mattermost.com/admin_console/team/5mouh8amgjnt8q5m14c7gte7pe/analytics), the metrics seem to stop after 21 users even though there are 57 users on the team.

For "Newly created users", for the [same team on ci-windows-mysql](https://ci-windows-mysql.mattermost.com/admin_console/team/5mouh8amgjnt8q5m14c7gte7pe/analytics), it looks like some newly created users aren't on the "recent active users" list. Are you sure it's pulling the users from the "recent active users" list?

jasonblais commented:
@it33 Also replying separately to you:

First, thanks for the review! You suggested changing section titles to be title cased, but they should be sentence case according to our style guide https://docs.mattermost.com/process/sg_mattermost-doc-style.html#capitalization

Also, the missing three stats are actually included under "Advanced system statistics (Enterprise)"

grundleborg commented:
@jasonblais so as far as I can figure out from the code, those two lists of users are populated from a single query, which loads the 100 most recently active users who are members of the team concerned.

On the client side that data is further transformed into 2 separate lists:

1. The recently active users list is generated by selecting just the 20 most recently active users from the list returned from the server (although due to a bug in how it's implemented it actually returns 21).
2. The newly created users list is generated by selecting the 20 newest users from the list returned from the server (with the same implementation bug meaning that actually up to 21 will be shown). This means it may be incorrect on busy teams where the number of users exceeds 100, as the newest 21 users may not be included in the 100 most recently active. I assume this is implemented this way as a hack to avoid a second query being necessary.

coreyhulen commented:
I think @grundleborg #2 is just a bug.  It should be a separate query.  I'm guessing when we moved to paging users we changed this.  I think before we had all the users so we would just loop through them all.

Also not sure it makes sense for this stat on the team lvl.  As an system admin I don't care about what team your in, but I would care about most recently created accounts in the system.  But I guess this is a different issue from the docs.

jasonblais commented:
I created a new pull request addressing all the feedback from here with new updates for 3.6: https://github.com/mattermost/docs/pull/769

Regarding the issue with #2, I'll queue to PM channel and discuss whether the "recently created accounts" is appropriate at the team level anymore. I'll then create a ticket to fix the bug as well
